Nav Unification: left nav geometries

cpapazoglou opened the pull request:
#### Changes proposed in this Pull Request

* Make all the changes needed so that left nav looks (not works) like the wp-admin (spacing, margin, padding perspective) and basic colors
![](https://cln.sh/Ml1qtx+)

#### Testing instructions
* Apply D49204-code
* ~Apply D49005-code~ Now merged.
* Apply sticker to your sandbox `wp blog-stickers add --sticker=nav-unification --who=<username> --blog_id=<blog_id>`
* Run calyspo and add `?flags=nav-unification` to the url
* Go to your sandboxed site, it should look like per spacing, margin, padding perspective
![](https://cln.sh/vR9V7J+)

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->


Fixes #45460

cpapazoglou commented:
Current State
![](https://cln.sh/xtp9Gy+)

wp-desktop reviewed (dismissed):
WordPress Desktop CI Failure for job "wp-desktop-mac".

@cpapazoglou please inspect this job's build steps for breaking changes at [this link](https://circleci.com/gh/Automattic/wp-calypso/884024). For temporal failures, you may try to "Rerun Workflow from Failed".

Please also ensure this branch is rebased off latest Calypso.

mreishus commented:
Seeing some odd behavior on narrow browser windows
![2020-09-10_15-34](https://user-images.githubusercontent.com/937354/92801364-270e5880-f37b-11ea-88fe-b7c9653d76de.png)

cpapazoglou commented:
> Seeing some odd behavior on narrow browser windows
> ![2020-09-10_15-34](https://user-images.githubusercontent.com/937354/92801364-270e5880-f37b-11ea-88fe-b7c9653d76de.png)

Thanks @mreishus for your heads up! Fixed the bug here fa6f121

getdave reviewed (commented):
I took a look at this today. First impressions were "wow this looks really good they are almost the same". Nice job!

I did however spot some inconsistencies. Forgive me if I'm going over old ground or questions that were already answered. Here's what I noticed:

## Navs initial loading state

Calypso is on the left and WPAdmin on the right. It's mainly:

* missing spacers
* differing text colors
* differing icon colors

I guess the key difference is that we don't have the "Switch Site" toggle. Until that's in place these two experiences will seem quite different.

![Screen Shot 2020-09-22 at 15 26 37](https://user-images.githubusercontent.com/444434/93896118-755f1800-fce8-11ea-8bac-fba440a4f41d.png)

## Expanded Nav Item

I used `Settings` as the comparison. It looks to me like the font sizes don't match correctly.

Also the padding/spacing on the word `Settings` (even without the icon) seems inconsistent.

![Screen Shot 2020-09-22 at 15 28 40](https://user-images.githubusercontent.com/444434/93896393-c66f0c00-fce8-11ea-961e-96f1f0977df7.png)


Overall this is heading in the right direction. I would encourage you to ship something as early as you can as you could be tweaking this for a long time and I suspect that having a long running PR is going to get really uncomfortable for you really quickly.

I'd be happy enough to merge this first pass and then raise issues for all the items identified.

cpapazoglou commented:
@getdave  Thanks for your review! I generally tried to have some minimum changes here (initially didn't even intend on changing the colors).

I will try to address your feedback 1 -1.
Designs: https://www.figma.com/file/9cy0DIowQkPAu0rodb9jsL/WP-Admin-for-Dotcom?node-id=325%3A3688&viewport=1878%2C1527%2C0.7836900949478149

Site switcher -> https://github.com/Automattic/wp-calypso/issues/45574
Lighter text color -> Actually I refreshed the color with `#a2aab2` as per design
Word view -> Actually I removed it as per design (I suppose due to narrower width)
Icons -> https://github.com/Automattic/wp-calypso/issues/45844
Separator -> it seems that something has changed.. Fixed
Collapse button -> https://github.com/Automattic/wp-calypso/issues/45674
font size -> fixed 
settings -> I think it is the icon missing that gives this feeling. Let's iterate again after icon is placed

I rebased with master

getdave reviewed (approved):
I re-tested this and it's pretty darn close to 1:1 with Calypso and WPAdmin now. IMGO it's worth merging and iterating from here.

I would say that this PR appears to touch quite a few CSS files that I'm unfamiliar with and thus I'm not sure what the knock on effects could be (eg: `client/components/domains/trademark-claims-notice/style.scss`). That said, switching out Sass vars for CSS native vars seems harmless enough.

Merge and iterate.

cpapazoglou commented:
> I re-tested this and it's pretty darn close to 1:1 with Calypso and WPAdmin now. IMGO it's worth merging and iterating from here.
> 
> I would say that this PR appears to touch quite a few CSS files that I'm unfamiliar with and thus I'm not sure what the knock on effects could be (eg: `client/components/domains/trademark-claims-notice/style.scss`). That said, switching out Sass vars for CSS native vars seems harmless enough.
> 
> Merge (with the caveat you need to get those tests 🟢 ) and then iterate from here.
> 
> Great work 🥇

Thanks Dave, indeed I needed to refactor some var calculations in order that tests pass! Thanks god we have tests 🎉 !

mreishus commented:
We reverted this due to an issue caused on prod.  With the feature off, if you selected a site from the site switcher, the site switcher would remain on the screen (covering up the blog menu) instead of disappearing.

I looked into it briefly, and think it's related to negative css variables like: `-var`.  Example:

![2020-09-23_16-55](https://user-images.githubusercontent.com/937354/94074383-c3464f80-fdbe-11ea-92eb-de4e17a7796f.png)

Changing the value in my browser tools to `translateX(calc(-1 * var(--sidebar-width-max)))` seemed to resolve the error.

cpapazoglou commented:
@mreishus thanks for the heads up and the investigation! Yes, it seems that css vars need a calc to work (in contrast to scss vars).
